lib/settings.php

```diff
@@ -14,26 +14,38 @@
 require_once(dirname(realpath(__FILE__)) . '/ReportConfigProcessor.php');
 
 if (isset($_GET['action'])) {
-    if ($_GET['action'] === 'saveReportSummary') {
-        $requestBody = trim(file_get_contents('php://input'));
+    $requestBody = trim(file_get_contents('php://input'));
 
-        if (!empty($requestBody)) {
-            $data = json_decode($requestBody, true);
+    if (!empty($requestBody)) {
+        $data = json_decode($requestBody, true);
+
+        if ($_GET['action'] === 'saveReportSummary') {
             $reportSummary = json_decode($data['reportSummary'], true);
 
             $reportConfig = new ReportConfig($project_id, $module);
             $result = $reportConfig->saveReportSummary($reportSummary['reportSummary']);
 
             if ($result === true) {
-                $report = json_decode(\REDCap::getReport($reportSummary['reportSummary']['reportId'], 'json', 'export', true /* export labels */), true);
+                $report = json_decode(\REDCap::getReport($reportSummary['reportSummary']['reportId'], 'json', true /* export labels */), true);
                 $reportProcessor = new ReportConfigProcessor($report, $reportSummary['reportSummary']);
 
                 exit(json_encode(array($reportProcessor->summaryConfig())));
-            } else {
-                http_response_code(400);
-                exit(json_encode($result));
+            }
+
+        } else if ($_GET['action'] === 'saveReportSummaries') {
```

Comment:
`saveReportSummaries` was added but I also refactored this code a bit - mainly to prevent duplication.